src/status_im/contexts/chat/home/new_chat/view.cljs

```diff
@@ -49,13 +49,27 @@
                                (rf/dispatch [:open-modal :new-contact]))}
       (i18n/label :t/add-a-contact)]]))
 
+(def ^:private contacts-selection-limit (dec constants/max-group-chat-participants))
+
 (defn contact-item-render
   [_]
   (fn [{:keys [public-key] :as item}]
-    (let [user-selected? (rf/sub [:is-contact-selected? public-key])
-          on-toggle      #(if user-selected?
-                            (re-frame/dispatch [:deselect-contact public-key])
-                            (re-frame/dispatch [:select-contact public-key]))]
+    (let [user-selected?          (rf/sub [:is-contact-selected? public-key])
+          selected-contacts-count (rf/sub [:selected-contacts-count])
+          on-toggle               (fn []
```

Comment:
`on-toggle` here could have been refactored into a separate event. To keep the UI simpler and the logic of the handler easily unit testable :)

Reply:
Also, components will not get triggered by `:selected-contacts-count` sub, as this can be easily accessible in event. Great suggestion :clap: